src/main/scanLogic/scanRunners/analyzerManager.ts

```diff
@@ -148,6 +151,33 @@ export class AnalyzerManager {
         };
     }
 
+    private isTokenValidationEnabled(): string {
+        let xraySemver: semver.SemVer = new semver.SemVer(this._connectionManager.xrayVersion);
+        if (xraySemver.compare(DYNAMIC_TOKEN_VALIDATION_MIN_XRAY_VERSION) < 0) {
+            this._logManager.logMessage(
+                'You cannot use dynamic token validation feature on xray version ' +
+                    this._connectionManager.xrayVersion +
+                    ' as it requires xray version ' +
+                    DYNAMIC_TOKEN_VALIDATION_MIN_XRAY_VERSION,
+                'INFO'
+            );
+            return 'false';
+        }
+        if (Configuration.enableTokenValidation()) {
+            return 'true';
+        }
+        let response: Promise<boolean> = this._connectionManager.isTokenValidationPlatformEnabled();
+        let tokenValidation: boolean = false;
+        response.then(res => {
+            tokenValidation = res;
+        });
```

Comment:
```suggestion
        let tokenValidation: boolean = await this._connectionManager.isTokenValidationPlatformEnabled();
```

Reply:
This enforces the function to be async no ?

Reply:
using `async` `await` is just cleaner method than using `then`

Reply:
the code you are commenting on is outdated